File: src/main/java/com/gisiona/demodevinia/infrastructure/adapter/persistence/UserJpaRepository.java
package com.gisiona.demodevinia.infrastructure.adapter.persistence;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserJpaRepository extends JpaRepository<UserEntity, Long> {
    
    Optional<UserEntity> findByEmail(String email);
    
    boolean existsByEmail(String email);
}

File: src/main/kotlin/com/financial/control/infrastructure/adapter/persistence/UserJpaRepository.kt
package com.financial.control.infrastructure.adapter.persistence

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserJpaRepository : JpaRepository<UserEntity, Long> {
    fun findByEmail(email: String): Optional<UserEntity>
    fun findByCpf(cpf: String): Optional<UserEntity>
    fun existsByEmail(email: String): Boolean
    fun existsByCpf(cpf: String): Boolean
    
    @Query("SELECT u FROM UserEntity u WHERE u.active = true")
    fun findAllActiveUsers(): List<UserEntity>
    
    @Query("SELECT u FROM UserEntity u WHERE u.id = :id AND u.active = true")
    fun findActiveUserById(id: Long): Optional<UserEntity>
}
